client/lib/tree-select/README.md

```diff
@@ -0,0 +1,70 @@
+# `treeSelect`
+
+This module exports a function which creates a cached state selector for use with the Redux global application state. It is a good idea to use this function over plain selectors whenever either the computation over state or React's rerenders are expensive.
+It is called `treeSelect` because it internally uses a [WeakMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap) based dependency tree in order to allow the gc to free memory without explicitly clearing the cache.
```

Comment:
> It is called `treeSelect` because it internally uses a [WeakMap]

That doesn't make sense. I think we're missing the logical warrant or we probably just need to remove the reference to `WeakMap` or add it later when it's not tied to `tree`

Reply:
read a few more words and then I think it makes sense:

> It is called `treeSelect` because it internally uses a WeakMap based dependency _tree_. 

I can switch around the wording so that it reads something like:

> it is called `treeSelect` because it internally uses a tree of dependencies in order to allow the gc to free memory without explicitly clearing the cache.

Reply:
> read a few more words and then I think it makes sense:

I read those words and the sense never came

> Something like:

Removing the `WeakMap` mention as you did makes sense.